tools/tertiary-analysis/pyscenic/pyscenic_aucell.xml

```diff
@@ -0,0 +1,108 @@
+<tool id="pyscenic_aucell" name="PySCENIC AUCell" profile="21.09" version="0.12.1+galaxy0">
+    <description>calculates AUCell to find relevant regulons/gene sets</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <command><![CDATA[
+        ln -s '${expression_mtx_fname}' expr_mat.loom &&
+        ln -s '${signatures_fname}' regulons.tsv &&
+
+        pyscenic aucell expr_mat.loom regulons.tsv
+           -o aucell.tsv
+        #if $transpose
+           -t
+        #end if
+        #if $weights
+           -w
+        #end if
+       --num_workers \${GALAXY_SLOTS:-1}
+        #if $seed
+           --seed '${seed}'
+        #end if
+        #if str($rank_threshold):
+           --rank_threshold '${rank_threshold}'
+        #end if
+        #if $auc_threshold
+           --auc_threshold '${auc_threshold}'
+        #end if
+        #if $nes_threshold
+           --nes_threshold '${nes_threshold}'
+        #end if
+        #if $cell_id_attribute
+           --cell_id_attribute '${cell_id_attribute}'
+        #end if
+        #if $gene_attribute
+           --gene_attribute '${gene_attribute}'
+        #end if
+        $sparse
+
+        && mv aucell.tsv '${output}'
+    ]]></command>
+    <inputs>
+        <param name="expression_mtx_fname" format="loom" type="data" label="Expression Matrix File" help="The file that contains the expression matrix for the single-cell experiment. Supported formats: csv (rows=cells x columns=genes) or loom (rows=genes x columns=cells)."/>
+        <param name="signatures_fname" type="data" format="csv,tabular" label="Gene Signatures/Regulons File" help="The file that contains the gene signatures (usually the precomputed regulons). Currently only csv/tsv supported, could be extended."/>
+        <param name="transpose" type="boolean" label="Transpose Expression Matrix" help="Transpose the expression matrix if supplied as csv (rows=genes x columns=cells)." optional="true"/>
+        <param name="weights" type="boolean" label="Use Weights for Recovery Analysis" help="Use weights associated with genes in recovery analysis. Relevant when gene signatures are supplied as json format." optional="true"/>
+        <param name="seed" type="integer" label="Seed for Ranking" help="Seed for the expression matrix ranking step. The default is to use a random seed." optional="true"/>
+        <param name="rank_threshold" type="integer" label="Rank Threshold" help="The rank threshold used for deriving the target genes of an enriched motif (default: 5000)." optional="true"/>
+        <param name="auc_threshold" type="float" label="AUC Threshold" help="The threshold used for calculating the AUC of a feature as fraction of ranked genes (default: 0.05)." optional="true"/>
```

Comment:
add min/max if you can to all floats/int

Reply:
A great suggestion, but I'm going to pass on this one, goes beyond the amount of time I can put to this. If anyone wants it dearly, the contribution will be welcome.